includes/admin.form.inc

```diff
@@ -127,23 +132,25 @@ function islandora_video_admin_submit($form, &$form_state) {
  * Form validation ensuring the selected ffmpeg options are available.
  */
 function islandora_video_admin_validate($form, &$form_state) {
-  // Ensure the mp4 audio codec is enabled.
-  $raw_value = $form_state['values']['islandora_video_mp4_audio_codec'];
-  if ($raw_value) {
-    if (preg_match('/[^0-9a-zA-Z_-]/', $raw_value) === 1) {
-      form_set_error('', "The value entered for MP4 audio codec contains forbidden characters.");
-      return;
+  // Ensure the mp4 audio codec is present if MP4 derivative is enabled.
+  if ($form_state['values']['islandora_video_make_mp4_locally'] == TRUE) {
+    $raw_value = $form_state['values']['islandora_video_mp4_audio_codec'];
+    if ($raw_value) {
+      if (preg_match('/[^0-9a-zA-Z_-]/', $raw_value) === 1) {
+        form_set_error('islandora_video_mp4_audio_codec', "The value entered for MP4 audio codec contains forbidden characters.");
+        return;
+      }
+      $ffmpeg = ($form_state['values']['islandora_video_ffmpeg_path'] !== variable_get('islandora_video_ffmpeg_path', 'ffmpeg') ? $form_state['values']['islandora_video_ffmpeg_path'] : variable_get('islandora_video_ffmpeg_path', 'ffmpeg'));
+      $safe_value = strtolower(trim($raw_value));
+      $command = $ffmpeg . ' -encoders 2>/dev/null | grep "^ ...... ' . $safe_value . ' "';
+      exec($command, $output, $ret);
+      if (!$output) {
+        form_set_error('islandora_video_mp4_audio_codec', 'The selected MP4 codec was not found in ffmpeg. Try using aac or enable the desired codec.');
```

Comment:
@rosiel question: I see if the preg match fail we set the error and return (good) but if the command is bad (no output) we rewrite the islandora_video_mp4_audio_codec field value anyway with our cleaned one and don't explicitly return. Is that expected behavior? Rest looks good.

Reply:
I didn't change this logic, but that's what I think was intended. A validate handler usually doesn't return until it's finished validating all the fields, so I'm inclined to think the above `return` is the one that's dubious. Though in this case, we're only validating one field (codec) so that doesn't matter. 

If the problem is that the path to ffmpeg is wrong, it's true, we don't differentiate where the error is and we would mark the wrong field as failing. I'll fix.

Reply:
In latest commit I addressed this. Now, we hard return from the validation function if either the codec or ffmpeg path* contain forbidden characters, then see if ffmpeg -version runs (if not, the problem is with the ffmpeg path) and if that works then test if the codec is present. That way the appropriate box will be marked invalid.

* Eventually there'll be a PR for ISLANDORA-1999 that will change this logic altogether, but that hasn't been merged yet, and this validation error seemed isolated enough to fix on its own.

Reply:
@DiegoPino does that address your concerns?